### PR TITLE
Issue #538 - Adding bash utility

### PR DESCRIPTION
**Pull request by jasonrahm**
Updated file:
f5/bigip/tm/util/**init**.py

Added files:
f5/bigip/tm/util/Bash.py
f5/bigip/tm/util/test/test_bash.py
test/functional/tm/util/test_bash.py

### CODE REVIEW

**Comment by jasonrahm** [comment]
@caphrim007 and @zancas this is ready for review. 

Structurally I believe it is sound, but I wonder if we should allow any string in utilCmdArgs that doesn't start with "-c ..." since any other use of bash from REST would be useless. Thoughts appreciated.

**Comment by jasonrahm** [comment]
@pjbreaux & @zancas, since you're in the neighborhood today, and thoughts on my previous comment?

**Comment by pjbreaux** [comment]
What would happen if I sent in FakeiControl.exec_cmd('run', utilsCmdArgs='env')?

**Comment by jasonrahm** [comment]
likely bad things.

**Comment by jasonrahm** [comment]
The actual answer is:

"/bin/env: /bin/env: cannot execute binary file "

So I'm guessing I should back off from testing for /bin/bash and just test for starting with /bin/ to catch any other unacceptable arguments passed to bash?

**Comment by jasonrahm** [comment]
@jgruber requested the functionality and really only needs support for bash -c.....

But if there are other important use cases we need to allow, I want to be thorough.

**Comment by pjbreaux** [comment]
I'm for adding the restriction if it will help protect users and notify them of the requirement. And when we get a user that needs to do something without -c, then we can handle it then.

**Comment by jasonrahm** [comment]
Ok, I'll rework the code and resubmit tomorrow. Thanks for the feedback.

**Comment by jasonrahm** [comment]
@caphrim007 this is ready for review.

**Comment by jasonrahm** [comment]
looks like I'm short a test case in coveralls @caphrim007, I'll investigate.

**Comment by jasonrahm** [comment]
ok, @caphrim007, ready for review.

File: f5/bigip/tm/util/Bash.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+from f5.bigip.resource import UnnamedResource
+from f5.utils.util_exceptions import UtilError
+
+
+class Bash(UnnamedResource, CommandExecutionMixin):
+    """BIG-IP® utility command
+
+    .. note::
+
+        This is an unnamed resource so it has not ~Partition~Name pattern
```
Comment (caphrim007) [inline f5/bigip/tm/util/Bash.py:39]: "no" instead of "not". We will fix this in a follow-up PR

File: test/functional/tm/util/test_bash.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        utilCmdArgs='-c "echo hello >> /var/tmp/test.txt"')
+
+    # commandResult should not be present if this was successful
+    assert 'commandResult' not in bash1.__dict__
+
+    bash2 = mgmt_root.tm.util.bash.exec_cmd(
+        'run',
+        utilCmdArgs='-c df -k')
+
+    # commandResult should b present with data from 'df -k'
```
Comment (caphrim007) [inline test/functional/tm/util/test_bash.py:42]: "be". Again, we'll fix it in a follow-up PR

**Review by caphrim007 — APPROVED** [review]
lgtm

**Comment by jasonrahm** [comment]
thanks, I'll fix those with next command in util.

**Pull request merged**